File: crates/klyron_config/bindings/cpp/builder.cpp
#include "builder.hpp"

namespace klyron::_config {

ConfigBuilder::ConfigBuilder() : config_{} {}

ConfigBuilder& ConfigBuilder::set_config(const ConfigConfig& cfg) {
    config_ = cfg;
    return *this;
}

ConfigConfig ConfigBuilder::config() const {
    return config_;
}

}

File: crates/klyron_config/bindings/rust/builder.rs
//! Builder for klyron_config
use crate::types::ConfigConfig;

pub struct ConfigBuilder {
    config: ConfigConfig,
}

impl ConfigBuilder {
    pub fn new() -> Self {
        Self { config: ConfigConfig::default() }
    }
    pub fn with_version(mut self, version: &str) -> Self {
        self.config.version = version.to_string();
        self
    }
    pub fn build(self) -> ConfigConfig {
        self.config
    }
}
